app/routes.js

```diff
@@ -46,43 +43,50 @@ module.exports = function(app) {
             .exec(resHandler(res));
     });
 
+    // get specific field from screen document
+    app.get('/api/screen/:id/:field', function(req, res) {
+        Screen
+            .find({'_id': req.params.id})
+            .select(req.params.field)
+            .exec(resHandler(res));
+    });
+
     // get all samples belonging to screen
     app.get('/api/samples/:screen', function(req, res) {
         Sample
         .find({ 'screen': req.params.screen })
-        .select('_id row column plate gene_name feature_vector_std ' +
+        .select('_id row column plate gene_name ' +
                 'pca_vector neighbours')
         .exec(resHandler(res));
     });
 
-    // get specific sample
-    app.get('/api/sample/:id', function(req, res) {
-          Sample
-            .find({ '_id': req.params.id })
-            .exec(resHandler(res))
+    // get value from sample document
+    app.get('/api/sample/:id/:field', function(req, res) {
+      Sample
+        .find({ '_id': req.params.id })
+        .select(req.params.field)
+        .exec(resHandler(res))
```

Comment:
Hey look, these are exactly the same lines as 49-51! -> Refactor. =) I see that "sample" and "screen" are different, but you can parameterise on that. Approximate example:

```javascript
get_field_callback = function(collection) {
    return function(req, res) {
        collection.find({}).select().exec() }

app.get('/api/sample/:id/:field', get_field_callback(Sample))`
```